src/components/DataTypes/oneOf.tsx

```diff
@@ -35,62 +22,28 @@ interface OneOfFilter extends JSONSchema {
 	};
 }
 
-interface DecodeFilterResult {
-	field: string;
-	operator: OperatorSlug;
-	value: any;
-}
-
-export const decodeFilter = (
-	filter: OneOfFilter,
-): DecodeFilterResult | null => {
-	const operator = filter.title;
-	if (!filter.properties) {
-		return null;
-	}
-
-	const keys = Object.keys(filter.properties);
-	if (!keys.length) {
-		return null;
-	}
-	let value: string;
-
-	const field = keys[0];
-
-	if (operator === 'is') {
-		value = filter.properties[field].const;
-	} else if (operator === 'is_not') {
-		value = filter.properties[field].not!.const;
-	} else {
-		return null;
-	}
-
-	return {
-		field,
-		operator,
-		value,
-	};
-};
-
 export const createFilter = (
+	key: string,
 	field: string,
-	operator: OperatorSlug,
+	operator: Operator,
 	value: any,
 	schema: JSONSchema,
+	recursive: boolean = false,
+	refScheme?: string,
 ): OneOfFilter => {
 	const { title } = schema;
-	const base: OneOfFilter = {
-		$id: randomString(),
-		title: operator,
-		description: getJsonDescription(
-			title || field,
-			operators[operator].getLabel(schema),
-			getCaption(value, schema) || '',
-		),
-		type: 'object',
-	};
+	const operatorSlug = operator.slug as OperatorSlug;
```

Comment:
Looks like `OperatorSlug` is always a string. Do we need to be creating the `type` in every file and setting it to the resolvable return type of the `operators` function? I think we can just use `string` as the type everywhere

Reply:
it is true that operatorSlug turns out to be a string, but by checking the type in my opinion it is clearer that it must be an operators key. string is so generic IMHO. if you think it is not understandable anyway, I can remove it and put only string.

Reply:
I hear your point, you're trying to establish the connection between the `operatorSlug` and the `operators`. To be honest I wouldn't create that extra type which is effectively just a `string` because seeing the extra type (as an outside observing the code) makes me think that we're expecting something other than a type we already have clear access to (i.e. `string | { value: string, id: number } | undefined`). That's what made this bit of code catch my eye was me wondering "it seemed like this just returns strings, what else is it expected to return?" I'd argue it makes things a little more confusing/convoluted

To be clear, if the typing ended up saying _exactly the keys expected in per case_, such as `'is' | 'is_not'` instead of just the generic `string` type, it would be 100% understandable to me and cool with me to have this extra `OperatorSlug` type created. So this would work in my opinion:
```
export const operators = () => ({
	is: 'is',
	is_not: 'is not',
} as const);

const operatorsObj = operators();

type OperatorSlug = keyof typeof operatorsObj;
```

Reply:
that would be a solution for sure, but this would not work for Object.ts case which is 

export const operators = (s: JSONSchema) => {

and operatorsObj cannot be called since `schema` is not defined. 

@thgreasi would you have any other options in mind ?